cpp/arcticdb/storage/config_cache.hpp

```diff
@@ -92,7 +82,15 @@ class ConfigCache {
             if(storage_conf_pos != storage_configs_.end())
                 storage_conf = storage_conf_pos->second;
 
-            storages.emplace_back(create_storage(path, mode, storage_conf));
+            if (auto it = native_storage_map.find(storage_name.value); 
+                storage_conf.config().Is<arcticdb::proto::s3_storage::Config>() && it != native_storage_map.end()) {
+                arcticdb::proto::s3_storage::Config s3_storage;
+                storage_conf.config().UnpackTo(&s3_storage);
+                storages.emplace_back(create_storage(path, mode, s3::S3Settings(it->second).load_from_proto(s3_storage)));
+            }
+            else {
+                storages.emplace_back(create_storage(path, mode, storage_conf));
```

Comment:
(nice to have) Assertion that the native config variant is monostate

Reply:
This hasn't been done?

Reply:
`NativeVariantStorage` is not a variant but an alias to `S3Settings` for now. It was named to be "variant" so later addition of other storages' native config could be easier